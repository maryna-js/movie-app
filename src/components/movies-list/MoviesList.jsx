import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { MovieCard } from "../movie-card/MovieCard";
import fetchMovies from '../../store/actions/fetchMovies';

export const MoviesList = () => {
    const dispatch = useDispatch();
    const { movies, isLoading, error } = useSelector(
        (state) => state.movies
    );

    useEffect(() => {
        dispatch(
            fetchMovies()
        )
    }, [dispatch]);

    return (
        <div className="w-full bg-white flex justify-center">
            <div className="border shadow-xl rounded-lg w-11/12 h-full p-4 m-4">
                {isLoading && (
                    <div className="text-sm text-gray-400 flex justify-center items-center">Loading..</div>
                )}
                {error && (
                    <div className="text-sm text-gray-400 flex justify-center items-center">
                        Sorry, app is not available
                    </div>
                )}
                {movies.map(({Poster, Type, Title, Year, imdbID}) => (
                    <MovieCard
                        poster={Poster}
                        type={Type}
                        title={Title}
                        year={Year}
                        id={imdbID}
                    />
                ))}
            </div>
        </div>
    );
};


