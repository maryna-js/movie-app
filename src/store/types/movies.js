export const LOADING = '/fetch/movies/LOADING';
export const LOADED = '/fetch/movies/LOADED';
export const ERROR = '/fetch/movies/ERROR';
