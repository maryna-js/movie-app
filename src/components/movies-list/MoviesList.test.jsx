import React from 'react';
import ReactDOM from 'react-dom';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import movies from '../../__mocks__/mockMovies.json';
import { MoviesList } from './MoviesList';

const mockStore = configureStore([]);

describe('movies list', () => {
  ReactDOM.createPortal = jest.fn((element) => element);
  const history = createMemoryHistory();
  history.push('/');
  const store = mockStore({
    movies: {
      movies,
      error: false,
      isLoading: false,
    },
  });
  store.dispatch = jest.fn();
  const getComponent = () =>
    render(
      <Provider store={store}>
        <Router navigator={history}>
          <MoviesList />
        </Router>
      </Provider>
    );
  it('Should render component', () => {
    expect(getComponent().asFragment()).toMatchSnapshot();
  });
  it('Should render with loading', () => {
    const store = mockStore({
      movies: {
        movies: [],
        error: false,
        isLoading: true,
      },
    });
    store.dispatch = jest.fn();
    const getComponent = () =>
      render(
        <Provider store={store}>
          <Router history={history}>
            <MoviesList />
          </Router>
        </Provider>
      );
    expect(getComponent().asFragment()).toMatchSnapshot();
  });
  it('Should render with error', () => {
    const store = mockStore({
      movies: {
        movies: [],
        error: true,
        isLoading: false,
      },
    });
    store.dispatch = jest.fn();
    const getComponent = () =>
      render(
        <Provider store={store}>
          <Router history={history}>
            <MoviesList />
          </Router>
        </Provider>
      );
    const { getByText } = getComponent();
    expect(getByText('Sorry, app is not available')).toBeTruthy();
  });
});
