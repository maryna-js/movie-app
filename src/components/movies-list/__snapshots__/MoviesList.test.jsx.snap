// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movies list Should render component 1`] = `
<DocumentFragment>
  <div
    class="w-full bg-white flex justify-center"
  >
    <div
      class="border shadow-xl rounded-lg w-11/12 h-full p-4 m-4"
    >
      <div
        class="flex flex-col items-start border rounded w-full h-full my-4 h-auto md:h-32"
      >
        <div
          class="flex flex-col md:flex-row justify-between items-center w-full"
        >
          <div
            class="flex w-full md:w-1/2"
          >
            <img
              alt="Star Wars"
              class="w-20 h-32 object-cover"
            />
            <div
              class="flex flex-col pl-7 pt-6"
            >
              <h2
                class="font-bold text-lg text-blue-800"
              >
                Star Wars
              </h2>
              <div
                class="text-blue-700 font-light text-sm pt-2"
              >
                movie
              </div>
              <div
                class="text-blue-600 font-light text-xs pt-2"
              >
                1977
              </div>
            </div>
          </div>
          <div
            class="w-64 flex items-center justify-center p-4"
          >
            <div
              class="mx-2"
            >
              <div />
            </div>
            <div
              class="text-lg font-medium text-blue-700 pl-4"
            >
              5.8 k
            </div>
            <div
              class="ml-8"
            >
              <div />
            </div>
            <div
              class="text-lg font-medium text-blue-700 pl-4 "
            >
              4.30
            </div>
          </div>
          <button
            class="text-blue-900 font-bold text-lg mr-6"
            data-testid="collapse"
          >
            +
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`movies list Should render with loading 1`] = `
<DocumentFragment>
  <div
    class="w-full bg-white flex justify-center"
  >
    <div
      class="border shadow-xl rounded-lg w-11/12 h-full p-4 m-4"
    >
      <div
        class="text-sm text-gray-400 flex justify-center items-center"
      >
        Loading..
      </div>
    </div>
  </div>
</DocumentFragment>
`;
